pkg/operator/ceph/cluster/controller.go

```diff
@@ -685,16 +701,16 @@ func (c *ClusterController) detectAndValidateCephVersion(cluster *cluster, image
 }
 
 func (c *ClusterController) handleUpdate(crdName string, cluster *cluster) (bool, error) {
-	c.updateClusterStatus(cluster.Namespace, crdName, cephv1.ClusterStateUpdating, "")
 
-	if err := cluster.createInstance(c.rookImage, cluster.Info.CephVersion); err != nil {
-		logger.Errorf("failed to update cluster in namespace %q. %v", cluster.Namespace, err)
+	config.ConditionExport(c.context, cluster.Namespace, crdName,
+		cephv1.ConditionUpdating, v1.ConditionTrue, "ClusterUpdating", "Cluster is updating")
+	if err := cluster.createInstance(c.rookImage, cluster.Info.CephVersion, c.status); err != nil {
+		logger.Errorf("failed to update cluster in namespace %q. %+v", cluster.Namespace, err)
 		return false, nil
 	}
-
-	c.updateClusterStatus(cluster.Namespace, crdName, cephv1.ClusterStateCreated, "")
-
-	logger.Infof("succeeded updating cluster in namespace %s", cluster.Namespace)
+	config.ConditionExport(c.context, cluster.Namespace, crdName,
+		cephv1.ConditionReady, v1.ConditionTrue, "ClusterUpdated", "Cluster Updated succesfully")
```

Comment:
nit: In the message only the first word should be upper case, if you could change this in all the messages

Reply:
Remind me, is there a timestamp associated with the conditions? It will be helpful to know when the operator completed the operation.

Reply:
Yes. There are two timestamps.
```
LastHeartbeatTime: When there is an update on the condition
LastTransitionTime: When condition changes from one state to another
```